apps/client/src/app/pages/db/db-comments/db-comments/db-comments.component.ts

```diff
@@ -64,37 +59,27 @@ export class DbCommentsComponent extends TeamcraftComponent implements OnInit {
 
   submitting = false;
 
-  showMoreComments$ = new BehaviorSubject<boolean>(false);
+  readonly showMoreComments$ = new BehaviorSubject<boolean>(false);
 
-  hasMoreComments$: Observable<number>;
+  readonly hasMoreComments$: Observable<number>;
 
-  constructor(private commentsService: DbCommentsService, private authFacade: AuthFacade, public translate: TranslateService,
-              @Inject(PLATFORM_ID) private platform: Object, private router: Router, private lazyData: LazyDataService,
-              private notificationService: NotificationService) {
+  constructor(
+    private commentsService: DbCommentsService,
+    private authFacade: AuthFacade,
+    public translate: TranslateService,
+    private router: Router,
+    private lazyData: LazyDataService,
+    private notificationService: NotificationService
+  ) {
     super();
-    this.user$ = this.authFacade.user$;
-    this.loggedIn$ = this.authFacade.loggedIn$;
-  }
 
-  ngOnInit() {
-    const contentComments = combineLatest([this.type$, this.id$]).pipe(
+    const comments$ = combineLatest([this.type$, this.id$, this.loggedIn$]).pipe(
+      filter(([type, id, loggedIn]) => !!type && !!id && !!loggedIn),
       switchMap(([type, id]) => {
-        return this.commentsService.getComments(`${type}/${id}`);
+        return combineLatest([this.commentsService.getComments(`${type}/${id}`), this.lang$]);
       }),
-      takeUntil(this.onDestroy$),
-      isPlatformServer(this.platform) ? first() : tap()
-    );
-
-    const lang$ = new BehaviorSubject<string>(this.translate.currentLang);
-
-    this.translate.onLangChange.pipe(
-      takeUntil(this.onDestroy$)
-    ).subscribe((change) => {
-      lang$.next(change.lang);
-    });
-
-    const comments$ = combineLatest([contentComments, lang$]).pipe(
       map(([comments, lang]) => {
+        console.log(comments, lang);
```

Comment:
Seems like there's a lot of remaining logs 😄

Reply:
🤦‍♀️ resolved.